src/fs/file.rs

```diff
@@ -338,50 +335,57 @@ impl File {
     /// }
     /// ```
     ///
-    /// [`ErrorKind::Interrupted`]: std::io::ErrorKind::Interrupted
     /// [`ErrorKind::UnexpectedEof`]: std::io::ErrorKind::UnexpectedEof
-    pub async fn read_exact_at<T: IoBufMut>(
+    pub async fn read_exact_at<T>(&self, buf: T, pos: u64) -> crate::BufResult<(), T>
+    where
+        T: BoundedBufMut,
+    {
+        let orig_bounds = buf.bounds();
+        let (res, buf) = self.read_exact_slice_at(buf.slice(..), pos).await;
```

Comment:
Instead of calling buf.slice(..) which I think still involves work, even when the compiler knows buf is an IoBufMut, could it call buf.slice_full()? Where slice_full is defined for both IoBuf and Slice<T> trivially? Or does the compiler already deduce all that from the '..' range? Would even the asserts be considered dead code?

Reply:
Indeed, it's possible to add such a method as an optimization, similarly to #53. Some of the asserts in an inlined `buf.slice(..)` call can be eliminated as dead code, but not all I think.